shell/common/rasterizer.cc

```diff
@@ -622,6 +623,10 @@ RasterStatus Rasterizer::DrawToSurfaceUnsafe(
 
     frame->set_submit_info(submit_info);
 
+    frame_timings_recorder.RecordRasterEnd(
```

Comment:
This actually might be an issue. It looks like above we're recording the presentation time - but this is actually happening way before any of the engine workload is executed. So we have no idea if we're going to hit or miss it.

Reply:
I guess you're right, it won't count any of the GPU time, right, just the preroll/paint in CPU? So either it can be reverted altogether, in which case the timings will include the swapBuffers call which might be backpressured, or I can try to find some way to move it deeper into the engine to call right before the swap?

Reply:
Not even GPU time, at this point we haven't even encoded the frame. I don't think this API could take GPU time into account, as we will "present" after encoding is finished but not necessarily after execution is complete

Reply:
I think how this worked before, is that the `ScopedFrame` destructor is what ended the raster time. And that wasn't destructed until after the frame was submitted and buffers swapped?

I don't know that much about the GPU terminology/model. What does encoded mean in this context?

With SurfaceTransaction there are some new APIs to get callbacks when the transaction is either committed and/or presented. Is it useful?